gridData/OpenDX.py

```diff
@@ -182,18 +192,118 @@ def write(self,file):
                       ('counts '+self.ndformat(' %d')) % tuple(self.shape))
 
 class array(DXclass):
-    """OpenDX array class"""
-    def __init__(self,classid,array=None,**kwargs):
+    """OpenDX array class.
+
+    See `Array Objects`_ for details.
+
+    .. _Array Objects:
+       https://web.archive.org/web/20080808140524/http://opendx.sdsc.edu/docs/html/pages/usrgu068.htm#Header_440
+    """
+    # equivalence between DX types and numpy dtypes.name
+    # (round-tripping is not guaranteed to produce identical types)
+    np_types = {
+        "uint8": "byte",         # DX "unsigned byte" equivalent
+        "int8": "signed byte",
+        "uint16": "unsigned short",
+        "int16": "short",         # DX "signed short" equivalent
+        "uint32": "unsigned int",
+        "int32": "int",           # DX "signed int"   equivalent
+        "uint64": "unsigned int", # not explicit in DX, for compatibility
+        "int64": "int",           # not explicit in DX, for compatibility
+        # "hyper",                # ?
+        "float32": "float",       # default
+        "float64": "double",
+        "float16": "float",       # float16 not available in DX, use float
+        # numpy "float128 not available, raise error
+        # "string" not automatically supported
+    }
+    dx_types = {
+        "byte": "uint8",
```

Comment:
is unsigned the default in DX files?

Reply:
no, float

Reply:
Sorry I meant is a byte really automatically unsigned?

Reply:
for compatibility we should here map `char` to an `uint8`. 

https://web.archive.org/web/20080808140524/http://opendx.sdsc.edu/docs/html/pages/usrgu068.htm#Header_440